src/Tests/RegistrationTest.php

```diff
@@ -25,19 +25,133 @@ class RegistrationTest extends WebTestBase {
+  protected function allowUserRegistration() {
+    $this->user_config
+      ->set('verify_mail', FALSE)
+      ->set('register', USER_REGISTER_VISITORS)
+      ->save();
+  }
+
+  /**
+   * Disable acquisition on registration.
+   */
+  protected function disableRegistrationAcquisition() {
+    $this->acquisition_config->set('acquisitions.registration', 0)->save();
+  }
+
+  /**
+   * Post user information to user register form.
+   *
+   * @param string $name
+   * @param string $mail_prefix
+   * @param string $pass
+   *
+   * @return array
+   *   Array of information that was used to create user.
+   */
+  protected function registerNewUser($name = '', $mail_prefix = '', $pass = '') {
+    $edit = array();
+    $edit['name'] = $name ? $name : $this->randomMachineName();
+    $edit['mail'] = $mail_prefix ? $mail_prefix . '@example.com' : $edit['name'] . '@example.com';
+
+    if (!is_null($pass)) {
+      $edit['pass[pass1]'] = $pass ? $pass : $this->randomMachineName();
+      $edit['pass[pass2]'] = $edit['pass[pass1]'];
+    }
+
+    $this->drupalPostForm('user/register', $edit, t('Create new account'));
+
+    return $edit;
+  }
+
+  /**
+   * Load User objects by an entity property.
+   *
+   * @param array $property
+   *   An array of a property value keyed by the property.
+   * @return array
+   *   An array of User entity objects indexed by their ids.
+   */
+  protected function getUsersByProperty(array $property) {
+    $user_storage = $this->container->get('entity_type.manager')->getStorage('user');
+    $accounts = $user_storage->loadByProperties($property);
+    return $accounts;
+  }
+
+
+  /**
    * Test the standard registration process when there is no existing user.
    */
   public function testNormalNone() {
-    // @todo: Change the site config to disable acquisitions.
+    // Set up test environment configuration,
+    $this->allowUserRegistration();
+    $this->disableRegistrationAcquisition();
+
+    // Test registering a new user when there are no existing users.
+    // Expected result: create a new user.
+    $edit = $this->registerNewUser();
+    $this->assertText(t('Registration successful. You are now logged in.'));
 
-    // @todo: Test when there is no user (should create).
+    // Load created user and check properties.
+    $accounts = $this->getUsersByProperty(['name' => $edit['name']]);
+
+    if (empty($accounts)) {
+      $this->fail('No User accounts loaded.');
+    }
+    else {
+      $account = reset($accounts);
+      $this->assertTrue($account->isActive());
+      $this->assertEqual($edit['name'], $account->getUsername());
+    }
   }
 
   /**
    * Test the acquisition registration process when there is no existing user.
    */
   public function testAcquisitionNone() {
-    // @todo: Test when there is no user (should create).
+    // Test registering a new user when there are no existing users.
+    // Expected result: create a new user.
+    $edit = $this->registerNewUser('', '', NULL);
```

Comment:
My previous comment appears to be wrong, though is there a reason we're doing it with NULL? Is it just because we haven't enabled unverified registration here? We don't need to test verified vs unverified in here, so I would suggest we make all the tests consistent (and the simplest). Not sure if that is disabling verification or always providing a password, possibly the former?